timemachine/potentials/generic.py

```diff
@@ -0,0 +1,236 @@
+from dataclasses import dataclass
+from typing import Any, Callable, Generic, Optional, TypeVar
+
+from typing_extensions import Protocol
+
+import timemachine.lib.potentials as gpu
+import timemachine.potentials.chiral_restraints as ref_chiral
+from timemachine.potentials import bonded as ref_bonded
+from timemachine.potentials import nonbonded as ref_nonbonded
+
+Array = Any
+Conf = Array
+Params = Array
```

Comment:
I went ahead and added a generic summed potential (and a reference implementation, which had been missing) in 7aee99a. 

> If it were added, I guess the Params type annotation here might become more generic, Union[Array, List[Array]] or something.

I think this is actually OK, since to be consistent with the GPU implementation the generic version needs to accept a flattened array of parameters (which is inconvenient for other reasons).